projects/packages/external-connections/src/class-external-connections.php

```diff
@@ -0,0 +1,272 @@
+			$connection = new Connection_Manager( 'jetpack' );
+			$status     = new Status();
+
+			if ( $status->is_offline_mode() ) {
+				return;
+			}
+
+			if ( ! $connection->has_connected_owner() ) {
+				return;
+			}
+
+			if ( ! $connection->is_user_connected() ) {
+				return;
+			}
+		}
+
+		add_settings_section(
+			'external_connections_section',
+			__( 'Integrations', 'jetpack-external-connections' ),
+			'__return_false',
+			$page
+		);
+
+		$asset_name = 'jetpack-external-connections-settings';
+		Assets::register_script(
+			$asset_name,
+			"build/$asset_name/$asset_name.js",
+			self::BASE_FILE,
+			array(
+				'in_footer'    => true,
+				'textdomain'   => 'jetpack-external-connections',
+				'dependencies' => array( 'wp-util' ),
+			)
+		);
+		Assets::enqueue_script( $asset_name );
+
+		$script_data = array();
+
+		foreach ( $services as $service ) {
+			if ( $host->is_wpcom_platform() ) {
+				$support_link = $service['support_link']['wpcom'];
+				if ( function_exists( 'localized_wpcom_url' ) ) {
+					$support_link = localized_wpcom_url( $support_link );
+				}
+			} else {
+				$support_link = Redirect::get_url( $service['support_link']['jetpack'] );
+			}
+
+			$is_connected = self::has_connection( $service['service'] );
+			$connect_url  = self::get_connect_url( $service['service'] );
+
+			add_settings_field(
+				'external_connections_field_' . $service['service'],
+				$service['title'],
+				function () use ( $service, $is_connected, $support_link ) {
+					?>
+					<div>
+						<button class="button-secondary jetpack-external-connection" type="button" data-service="<?php echo esc_attr( $service['service'] ); ?>">
+							<?php $is_connected ? esc_html_e( 'Disconnect', 'jetpack-external-connections' ) : esc_html_e( 'Connect', 'jetpack-external-connections' ); ?>
+						</button>
+						<p class="description">
+							<?php echo esc_html( $service['description'] ); ?>
+							<a href="<?php echo esc_url( $support_link ); ?>" target="_blank" data-target="wpcom-help-center"><?php esc_html_e( 'Learn more', 'jetpack-external-connections' ); ?></a>
+						</p>
+					</div>
+					<?php
+				},
+				$page,
+				'external_connections_section'
+			);
+
+			$script_data[ $service['service'] ] = array(
+				'isConnected' => $is_connected,
+				'connectUrl'  => $connect_url,
+				'deleteNonce' => wp_create_nonce( 'jetpack_delete_external_connection_' . $service['service'] ),
+			);
+		}
+
+		wp_add_inline_script(
+			$asset_name,
+			'const jetpackExternalConnectionsData = ' . wp_json_encode( $script_data ) . ';',
+			'before'
+		);
+	}
+
+	/**
+	 * Handles the AJAX request to delete an external connection.
+	 */
+	public static function ajax_delete_connection() {
+		if ( ! isset( $_REQUEST['service'] ) ) {
+			wp_die();
+		}
+
+		$service = sanitize_text_field( wp_unslash( $_REQUEST['service'] ) );
+		check_ajax_referer( 'jetpack_delete_external_connection_' . $service );
+
+		self::delete_connection( $service );
```

Comment:
Q: should we check if the operation was successful and return a response? 🤔 I'm mostly thinking about the Jetpack API request

Reply:
e7eab45881d6ffe59c4c2d18cecba5e74bf9e129